server/src/main/scala/services/graphstages/WorkloadGraphStage.scala

```diff
@@ -70,9 +70,14 @@ class WorkloadGraphStage(name: String = "",
       override def onPush(): Unit = {
         val timer = StageTimer(stageName, inFlightsWithSplits)
         val incomingFlights = grab(inFlightsWithSplits)
-        log.info(s"Received ${incomingFlights.flightsToUpdate.size} arrivals")
+        log.info(s"Received ${incomingFlights.flightsToUpdate.length} updated arrivals and ${incomingFlights.arrivalsToRemove.length} arrivals to remove")
 
-        val existingFlightTQMs: Set[TQM] = incomingFlights.flightsToUpdate.flatMap(fws => flightTQMs.getOrElse(fws.apiFlight.uniqueId, List())).toSet
+        val updatesTqms = incomingFlights.flightsToUpdate.flatMap(fws => flightTQMs.getOrElse(fws.apiFlight.uniqueId, List())).toSet
+        val removalUniqueIds = incomingFlights.arrivalsToRemove.map(_.uniqueId)
+        val removalTqms = incomingFlights.arrivalsToRemove.flatMap(a => flightTQMs.getOrElse(a.uniqueId, List())).toSet
+        val existingFlightTQMs: Set[TQM] = updatesTqms ++ removalTqms
+
+        flightTQMs --= removalUniqueIds
```

Comment:
might make sense to put this line below line 76 (or inline removalTqms) to make it easier to see what's going on.